sda/processor.go

```diff
@@ -10,6 +10,97 @@ import (
 	"github.com/dedis/cothority/network"
 )
 
+// Dispatcher is an interface whose sole role is to distribute messages to the
+// right Processor. No processing is done,i.e. no looking at packet content.
+// There are many ways to distribute messages: for the moment, only
+// BlockingDispatcher is implemented, which is a blocking dispatcher.
+// Later, one can easily imagine to have a dispatcher with one worker in a
+// goroutine or a fully fledged producer/consumers pattern in go routines.
+// Each Processor that want to receive all messages of a specific
+// type must register them self to the dispatcher using `RegisterProcessor()`.
+// The network layer must call `Dispatch()` each time it received a message, so
+// the dispatcher is able to dispatch correctly to the right Processor for
+// further analysis.
+type Dispatcher interface {
+	// RegisterProcessor is called by a Processor so it can receive any packet
+	// of type msgType.
+	// **NOTE** In the current version, if a subequent call RegisterProcessor
+	// happens, for the same msgType, the latest Processor will be used; there
+	// is no *copy* or *duplication* of messages.
+	RegisterProcessor(p Processor, msgType network.MessageTypeID)
+	/* // Dispatch will find the right processor to dispatch the packet to. The id*/
+	//// is the identity of the author / sender of the packet.
+	//// It can be called for example by the network layer.
+	//// If no processor is found for this message type, then an error is returned
+	/*Dispatch(packet *network.Packet) error*/
+}
+
+// Processor is an abstraction to represent any object that want to process
+// packets. It is used in conjunction with Dispatcher:
+// A processor must register itself to a Dispatcher so the Dispatcher will
+// dispatch every messages to the Processor asked for.
+type Processor interface {
+	// Process takes a ServerIdentity as the sender identity and the message
+	// sent.
+	Process(packet *network.Packet)
+}
+
+// BlockingDispatcher is a Dispatcher that simply calls `p.Process()` on a
+// processor p each time it receives a message with `Dispatch`. It does *not*
+// launch a go routine, or put the message in a queue, etc.
+// It can be re-uswored for more complex dispatcher.
+type BlockingDispatcher struct {
+	procs map[network.MessageTypeID]Processor
+}
+
+// NewBlockingDispatcher will return a freshly initialized BlockingDispatcher
```

Comment:
I don't know if we want to be a..l about making complete sentences in comments and end them with a '.'...